packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -275,10 +275,10 @@ export const KeyboardNavigationMixin = (superClass) =>
     }
 
     /** @private */
-    _ensureScrolledToIndex(index) {
+    __ensureFlatIndexInViewport(index) {
       const targetRowInDom = [...this.$.items.children].find((child) => child.index === index);
       if (!targetRowInDom) {
-        this.scrollToIndex(index);
+        this._scrollToFlatIndex(index);
```

Comment:
**note:** This method was missed to be updated after the signature of scrollToIndex was changed to support scrolling to hierarchical indexes.